plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2936,9 +2940,20 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
         String dependsOnTables = Optional.ofNullable(currentSnapshot)
                 .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_TABLES, ""))
                 .orElse("");
+        String dependsOnTableFunctions = Optional.ofNullable(icebergTable.currentSnapshot())
+                .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_FUNCTIONS, ""))
+                .orElse("");
+
         if (dependsOnTables.isEmpty()) {
+            // If the source of the MV is a PTF then source tables will be empty but source functions won't
+            if (!dependsOnTableFunctions.isEmpty()) {
+                Instant refreshTime = Optional.ofNullable(icebergTable.currentSnapshot().summary().get(TRINO_QUERY_START_TIME))
```

Comment:
Move the `refreshTime` variable, rather than have the logic twice.

Reply:
It fails in some tests when it is before `if (dependsOnTables.isEmpty()) ` this is why i put it there.

Reply:
When MV has not been refreshed yet, there is no snapshot so there is a null returned for currentSnapshot and thus this code fails. It works inside that if because `dependsOnTableFunctions` is not empty only when REFRESH already happened.
I don't like this repetition but I don't see any obvious way to get rid of it

Reply:
I can't only extract it to a separate method..

Reply:
ok im stupid

Reply:
refactored